docs/adrs/003-bugzilla-response-codes.md

```diff
@@ -0,0 +1,100 @@
+# Bugzilla disables webhooks after too many errors
+
+- Status: proposed
+- Date: 2023-27-11
+
+Tracking issue: 
+
+## Context and Problem Statement
+When bugzilla receives too many error responses from JBI, it stops triggering webhook calls for the entire project causing data to stop syncing. Frequently these errors are due to JBI being unable to process a payload due to errors in configuration (or incomplete configuration) in Jira or a mismatch of data for a single bug. 
+
+We don't want the entire sync process to stop because of this. We have identified four options to solve this problem.
+
+## Decision Drivers
+
+- Amount of initial engineering effort
+- Amount of maintenance effort
+- Overall performance of JBI (how quickly is data able to move)
+- How intuitive the solution is to the users that depend on the data (will picking the easiest option solve their needs?)
+
+## Considered Options
+For all of these options, we will be returning a succesful response to bugzilla's webhook calls. I propose a 202 status code to identify that we're accepting the request immediately and are processing it asynchronously.
+
+### Option 1: Log the failure and move on
+JBI will log that we couldn't process a specific payload, along with relevant ID's (bug id, jira ticket id, comment id, etc) so further investigation can be done if needed.
+
+Pros:
+- The simplest solution
+- Very low effort
+
+Cons: 
+- Will not alert people to data loss (without additional alerting functionality)
+- Still requires engineers to investigate further
+
+### Option 2: Ask a human to do something
+JBI will alert users that data could not be sync'd. This could happen through an IM alert or an email immediately, or a scheduled (daily?) report.
+
+### Option 3: Queue retries internally
+Create a persistence layer within the JBI containers that will queue and retry jobs for a specific length (2 hours?) of time. This could be done with an internal cache (redis) or database (postgres) within the container. After retries exceed the max time length, an error would be logged and the data would be dropped.
```

Comment:
Also, usually the issues are due to a Jira/JBI misconfiguration, and it is very unlikely that it will be resolved within a few hours. Sometimes it even takes days because we depend on Jira admins perms that we don't have